lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board
+  attr_accessor :cells
+  
+  def initialize  
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  # Displays current instance of board variable
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def update(input, player)
+    position = input.to_i-1
+    cells[position] = player.token
+  end
+
+ # Finally, a board can return values based on its state such as
+  def full?
+    cells.all?{ |i| i == "X" || i == "O"}
+  end
+
+  def turn_count
+    #turn_count based on how many positions in the cells array are filled.
+    cells.count{ |i| i == "X" || i == "O" }
+  end
```

Comment:
Excellent choice of iterators for both `full?` and `turn_count`. Very expressive.